### ISSUE TITLE
mangle breaks prod build

### ISSUE THREAD

**Issue by mastertinner** [AngularClass/angular2-webpack-starter#262]
As long as I leave `mangle: { screw_ie8 : true }` as it is set in `webpack.prod.config.js`, my `npm run build:prod` doesn't work. It builds fine but when I try to open the website in my browser (Chrome), I'm getting the following error: `EXCEPTION: No Directive annotation found on t`

Setting `mangle: false` fixes the problem.

**Comment by gdi2290** [comment]
this is a problem with Angular 2

**Comment by mastertinner** [comment]
Thanks, but then shouldn't we disable mangling until beta.2 is out?

**Comment by gdi2290** [comment]
@mastertinner can you make a pull-request and comment about it in prod webpack configs

**Comment by mastertinner** [comment]
Apparently this problem persists in beta.2...

**Comment by gdi2290** [comment]
yup there's still a problem with RouterLink when minified

**Comment by cristi-badila** [comment]
I stumbled into a similar issue while running an app I'm working on in production mode. When using the mangle names option, the outputed javascript results in angular generating a recursive `viewFactory` that will eventually throw a `Maximum call stack size exceeded` exception. I isolated the issue in a repository [here](https://github.com/cristi-badila/angular-mangle-issue/tree/master) which contains both the working non mangled js and the mangled js which results in angular throwing exceptions. Hope this helps.
